iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/lockfree_queue/index_queue.hpp

```diff
@@ -116,6 +116,7 @@ class IndexQueue
     ///    the default atomic constructor does not call the default constructor of the
     ///    underlying class.
     ///    See, http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0883r0.pdf
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays, hicpp-avoid-c-arrays)
```

Comment:
Please add a justification (and maybe remove the white spaces at the beginning of the comments above).

Reply:
Not needed with the current change to std::array, which is under debate.

Reply:
@FerdinandSpitzschnueffler As I undid the usage of `std::array` after discussion with @elBoberido I added the justification.

Note that I only use an extra line for the justification if it is too long to fit on the suppression command line. Now we may decide to do it differently in the future but I think this should be a separate PR once the details are decided. (I prefer to have this line optional).